projects/packages/videopress/src/class-access-control.php

```diff
@@ -0,0 +1,399 @@
+					$is_user_authed      = $restriction_details['can_access'];
+				}
+				break;
+		}
+
+		$this->update_video_restriction_details_on_wpcom( $guid, $restriction_details );
+
+		/**
+		 * Overrides video view authorization for current user.
+		 *
+		 * Example of making all videos public:
+		 *
+		 * function jp_example_override_video_auth( $is_user_authed, $guid ) {
+		 *  return true
+		 * };
+		 * add_filter( 'videopress_is_current_user_authed_for_video', 'jp_example_override_video_auth', 10, 2 );
+		 *
+		 * @param bool     $is_user_authed   The current user authorization state.
+		 * @param string   $guid             The video's unique identifier.
+		 * @param int|null $embedded_post_id The post the video is embedded..
+		 *
+		 * @return bool
+		 */
+		return $this->filter_is_current_user_authed_for_video( $is_user_authed, $guid, $embedded_post_id );
+	}
+
+		/**
+		 * Overrides video view authorization for current user.
+		 *
+		 * @param bool     $is_user_authed   The current user authorization state.
+		 * @param string   $guid             The video's unique identifier.
+		 * @param int|null $embedded_post_id The post the video is embedded..
+		 *
+		 * @return bool
+		 */
+	private function filter_is_current_user_authed_for_video( $is_user_authed, $guid, $embedded_post_id ) {
+		/**
+		 * Overrides video view authorization for current user.
+		 *
+		 * Example of making all videos public:
+		 *
+		 * function jp_example_override_video_auth( $is_user_authed, $guid ) {
+		 *  return true
+		 * };
+		 * add_filter( 'videopress_is_current_user_authed_for_video', 'jp_example_override_video_auth', 10, 2 );
+		 *
+		 * @param bool     $is_user_authed   The current user authorization state.
+		 * @param string   $guid             The video's unique identifier.
+		 * @param int|null $embedded_post_id The post the video is embedded..
+		 *
+		 * @return bool
+		 */
+		return (bool) apply_filters( 'videopress_is_current_user_authed_for_video', $is_user_authed, $guid, $embedded_post_id );
+	}
+	/**
+	 * Updates a video's privacy details on wpcom.
+	 *
+	 * @param string $guid    The video guid that needs updated privacy details.
+	 * @param array  $details The details.
+	 */
+	private function update_video_restriction_details_on_wpcom( $guid, $details = array() ) {
+		$video_blog_id   = $this->get_videopress_blog_id();
+		$args            = array(
+			'headers' => array( 'content-type' => 'application/json' ),
+			'method'  => 'POST',
+		);
+		$default_details = array(
+			'version'              => '1',
+			'provider'             => 'auth',
+			'title'                => 'Unauthorized',
+			'unauthorized_message' => 'Unauthorized',
+		);
+
+		$body = array_merge( $default_details, $details );
+
+		$endpoint = "sites/{$video_blog_id}/media/videopress-privacy-details/{$guid}";
+		$result   = Client::wpcom_json_api_request_as_blog( $endpoint, 'v2', $args, wp_json_encode( $body ), 'wpcom' );
```

Comment:
As this is called every time a user restrictions is checked, shouldn't it be linked to the user and use
```
wpcom_json_api_request_as_user
```

Reply:
removing this code entirely so things are more simple for now.